### ISSUE TITLE
Icons not truly centered 

### ISSUE THREAD

**Issue by Trulsmatias** [jlnrrg/simple_icons#11]
As with you I used https://github.com/NikSWE/flutter_brand_icons but ran in to the issue with null safety. Fortunately I came over your comment on an issue on that repo that led me to this repo https://github.com/NikSWE/flutter_brand_icons/issues/1#issuecomment-772851138. So I know you took inspiration from that repo. Unfortunately this repo is not centering the icons as the other repo did. This is not a breaking issue, but annoying. 

flutter_brand_icons:
<img width="135" alt="Skjermbilde 2021-04-04 kl  15 09 40" src="https://user-images.githubusercontent.com/21953916/113509803-ca150280-9557-11eb-8f36-5aa5ac33e867.png">

this repo:
<img width="132" alt="Skjermbilde 2021-04-04 kl  15 10 11" src="https://user-images.githubusercontent.com/21953916/113509814-db5e0f00-9557-11eb-97d9-f643b46fdaca.png">

**Comment by jlnrrg** [comment]
Both libraries just export IconData, so I highly doubt that this is an issue with the package.
If you'd like to provide an minimal example of the error you encounter, I will be more than happy to take a look :smile:

**Comment by Trulsmatias** [comment]
Thanks for the fast reply. I have had a lot on my plate lately and haven't had the chance to look into this before now. After doing some testing it looks like this only is a problem on web (which is where I am using the package). Here is a minimal example:

```dart
import 'package:flutter/material.dart';
import 'package:flutter_brand_icons/flutter_brand_icons.dart';
import 'package:simple_icons/simple_icons.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: Text("Icon Demo"),
        ),
        body: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: [
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                InkedIconButton(
                  icon: BrandIcons.apple,
                ),
                SizedBox(
                  width: 1,
                ),
                InkedIconButton(
                  icon: SimpleIcons.apple,
                ),
              ],
            ),
            SizedBox(
              height: 10,
            ),
            Row(
              mainAxisAlignment: MainAxisAlignment.center,
              children: [
                InkedIconButton(
                  icon: BrandIcons.googleplay,
                ),
                SizedBox(
                  width: 1,
                ),
                InkedIconButton(
                  icon: SimpleIcons.googleplay,
                ),
              ],
            )
          ],
        ),
      ),
    );
  }
}

class InkedIconButton extends StatelessWidget {
  final IconData icon;

  const InkedIconButton({Key key, this.icon}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return Ink(
      decoration: ShapeDecoration(
        color: Colors.black,
        shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.all(Radius.circular(10))),
      ),
      child: IconButton(
        iconSize: 35,
        icon: Icon(
          icon,
          color: Colors.white,
        ),
        onPressed: () {},
      ),
    );
  }
}
```

**Comment by jlnrrg** [comment]
Thanks for limiting the issue to flutter web.
Because the library just exports IconData and it does not differenciate between web or app, the only explanation I can think if is that there is an issue directly with the flutter web framework.
But I agree, that is is weird that such a problem did not occur with the brand_icon.
Maybe  @NikSWE (author of brand_icons) has an idea.

**Comment by Trulsmatias** [comment]
Let's hope he answers!

**Comment by jmarkham** [comment]
It's the font file.  Had the same issue with flutter web and html renderer.  Issue doesn't appear with canvaskit.  To validate it's something with the ttf file, I can download the SVGs I want from simpleicons.org, upload them to fluttericon.com, download the result, and use them as-is with the rendering as expected in both html and canvaskit rendering.

**Comment by NikSWE** [comment]
> It's the font file

I agree with @jmarkham's explanation, also @jlnrrg how are you converting the svg from simpleicons to ttf file? (I've used [icomoon](https://icomoon.io/app/#/select) to convert the svg's set to ttf file, manual process)

**Comment by NikSWE** [comment]
if we could use svg instead of ttf on flutter web that would be great, would suggest @jlnrrg to ask this on the flutter repo for better guidance.

**Comment by jlnrrg** [comment]
@NikSWE I only grab the ttf file from simpleicons directly (I have no idea how they build it)
with the ttf file I run `ttx -t cmap SimpleIcons.ttf` to get `ttx` file where I can grab the code from. (it is basicaly xml)

**Comment by NikSWE** [comment]
this way people can quickly get the name of the icon

**Comment by jlnrrg** [comment]
I think they added it later [github issue](https://github.com/simple-icons/simple-icons-font/issues/14)

I don't really see the benefit of it in regards to the effort, as people can just use that page or look in the documentation.
But if you want to build it, feel free to do so, and make a PR or publish it in a seperate repo, where we can link to then.

**Comment by NikSWE** [comment]
does this issue only happen in html renderer on web ?

**Comment by NikSWE** [comment]
did someone try this on any flutter desktop build ?

**Comment by NikSWE** [comment]
also pls try the above mentioned package and test it on web (html renderer)

**Comment by Trulsmatias** [comment]
> @Trulsmatias apart from the two icons mentioned which other icons are showing such difference ?

I only use those two. Have not checked other icons.

**Comment by jlnrrg** [comment]
@Trulsmatias @jmarkham in the latest version `6.1.0` the issue should have been fixed.
Can you please confirm, that the error is also resolved on your side?

**Comment by Trulsmatias** [comment]
This is how it looks like now. Red is SimpleIcons, blue is BrandIcons.

Versions in pubspec.yaml
```
  simple_icons: ^6.1.0
  flutter_brand_icons: 1.1.0
```

web - canvaskit. Blue looks correct as it gets all the way to the top edge
<img src="https://user-images.githubusercontent.com/21953916/152601449-10335ee9-c269-40de-a2a9-0fa0e7a4e0c1.png" width="300"/>

web - html. Even worse unfortunately.
<img src="https://user-images.githubusercontent.com/21953916/152601446-e18d012e-f982-4485-9ffe-d5a1558f3f25.png" width="300"/>

web - html - centered to better visualize the problem. Red goes outside the screen.
<img src="https://user-images.githubusercontent.com/21953916/152601444-8e1e3887-1460-401a-bc56-f4b302ac165b.png" width="300"/>

Double-checked on mobile and the problem was there too, so maybe the scope of this issue should be reverted back to including mobile again:

Android emulator. Blue gets all the way to the top, red doesn't.
<img src="https://user-images.githubusercontent.com/21953916/152682896-e20b8c8e-f2d1-443a-85b4-1554d93607e2.png" width="250"/>

iPhone simulator. This was interesting, on all the other images blue was higher than red. On iPhone this was reversed, so red overshoots outside the screen. Again blue reaches the edge as expected.
<img src="https://user-images.githubusercontent.com/21953916/152683812-031c19e8-ad6f-44e8-b162-12e362dd04c7.png" width="250"/>

Code I used:
```import 'package:flutter/material.dart';
import 'package:flutter_brand_icons/flutter_brand_icons.dart';
import 'package:simple_icons/simple_icons.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    const double iconSize = 400;
    return MaterialApp(
      home: Scaffold(
        body: Stack(
          children: [
            Icon(
              SimpleIcons.apple,
              color: Colors.red,
              size: iconSize,
            ),
            Icon(
              BrandIcons.apple,
              color: Colors.blue,
              size: iconSize,
            ),
          ],
        ),
      ),
    );
  }
}
```

**Comment by jlnrrg** [comment]
@Trulsmatias thank you for your feedback. I was able to reproduce the issue.

@NikSWE I tried your `flutter_brand_icons` package with the most recent SimpleIcons.ttf file and it had the same missalignment as mine. So the question is, if you created the [brand.ttf](https://github.com/NikSWE/flutter_brand_icons/blob/master/fonts/brands.ttf) of your package yourself or if it was just a renamed SimpleIcons.ttf of that year (3 years ago).

**Comment by NikSWE** [comment]
I think it would be better to raise this concern on the simpleicons repo, as this issue is in the font file. Could be that they have fixed this issue and not updated the font file.

**Comment by jlnrrg** [comment]
@NikSWE thanks for the fast answer.
Can you go a little bit more into detail if I am not correct?
I would assume it was like this:
1. download all simple-icons svgs
2. import them all in icomoon
3. export icomoon as font

**Comment by NikSWE** [comment]
Correct, this is the process I followed to create the TTF font file

**Comment by NikSWE** [comment]
you could try one thing just create another font file containing only the apple icon and then import that in the code to see the difference between all the three

**Comment by jlnrrg** [comment]
Because I am always searching for the correct issue:
I escalated the problem back then and now it is opened again at simple-icons.
https://github.com/simple-icons/simple-icons-font/issues/113
